In [1]:
import pandas as pd

In [2]:
videogamesales = pd.read_csv("./input/vgsales.csv")

In [3]:
import scipy.stats as st

# Sales Global

In [4]:
confidence = 0.90
st.t.interval(confidence, len(videogamesales['Global_Sales'])-1, loc=videogamesales['Global_Sales'].mean(), scale=st.sem(videogamesales['Global_Sales']))

(0.5175860248001194, 0.5572952862013607)

In [5]:
confidence = 0.95
st.t.interval(confidence, len(videogamesales['Global_Sales'])-1, loc=videogamesales['Global_Sales'].mean(), scale=st.sem(videogamesales['Global_Sales']))

(0.513781998985215, 0.5610993120162651)

In [6]:
confidence = 0.99
st.t.interval(confidence, len(videogamesales['Global_Sales'])-1, loc=videogamesales['Global_Sales'].mean(), scale=st.sem(videogamesales['Global_Sales']))

(0.5063466018870761, 0.568534709114404)

In [7]:
confidence = 0.9999
st.t.interval(confidence, len(videogamesales['Global_Sales'])-1, loc=videogamesales['Global_Sales'].mean(), scale=st.sem(videogamesales['Global_Sales']))

(0.49046946533976576, 0.5844118456617143)

# Sales Japan

In [8]:
confidence = 0.90
st.t.interval(confidence, len(videogamesales['JP_Sales'])-1, loc=videogamesales['JP_Sales'].mean(), scale=st.sem(videogamesales['JP_Sales']))

(0.07383263067996233, 0.08173069020205982)

In [9]:
confidence = 0.95
st.t.interval(confidence, len(videogamesales['JP_Sales'])-1, loc=videogamesales['JP_Sales'].mean(), scale=st.sem(videogamesales['JP_Sales']))

(0.07307602072929802, 0.08248730015272414)

In [10]:
confidence = 0.99
st.t.interval(confidence, len(videogamesales['JP_Sales'])-1, loc=videogamesales['JP_Sales'].mean(), scale=st.sem(videogamesales['JP_Sales']))

(0.07159714132486526, 0.08396617955715689)

In [11]:
confidence = 0.9999
st.t.interval(confidence, len(videogamesales['JP_Sales'])-1, loc=videogamesales['JP_Sales'].mean(), scale=st.sem(videogamesales['JP_Sales']))

(0.06843922387522955, 0.0871240970067926)

# Sales With 10% of rows

In [12]:
def sample_first_prows(data, perc=0.1):
    return data.head(int(len(data)*(perc)))

In [13]:
smallvideogamesales = sample_first_prows(videogamesales)

In [14]:
confidence = 0.95
st.t.interval(confidence, len(smallvideogamesales['Global_Sales'])-1, loc=smallvideogamesales['Global_Sales'].mean(), scale=st.sem(smallvideogamesales['Global_Sales']))

(2.971612774483198, 3.3556566649381505)

In [15]:
confidence = 0.95
st.t.interval(confidence, len(smallvideogamesales['JP_Sales'])-1, loc=smallvideogamesales['JP_Sales'].mean(), scale=st.sem(smallvideogamesales['JP_Sales']))

(0.356597361652292, 0.4390144526936945)

Je ziet dat naarmate het confidence level toeneemt de interval groter wordt

Je kunt ook zien dat de interval met maar 10% van de data een stuk groter is dan dat hij de hele dataset gebruikt
Bij Global sales is dit (0.513781998985215, 0.5610993120162651) bij 100% en dit (2.971612774483198, 3.3556566649381505) bij 10% hetzelfde geld voor sales in Japan